pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+
+            Self::deposit_event(Event::<T>::WithdrawnFromUnregistered {
+                who: staker,
+                amount: staked_value,
+            });
+
+            Ok(().into())
+        }
+
+        // Not sure here whether to calculate total rewards for all creators
+        //  or to withdraw per-creator rewards (preferably)
+        #[pallet::call_index(7)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn claim_staker_reward(origin: OriginFor<T>, space_id: SpaceId, restake: bool) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            // Ensure we have something to claim
+            let mut staker_info = Self::staker_info(&staker, space_id);
+            let (era, staked) = staker_info.claim();
+            ensure!(staked > Zero::zero(), Error::<T>::NotStakedCreator);
+
+            let creator_info =
+                RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+            if let CreatorState::Unregistered(unregister_era) = creator_info.state {
+                ensure!(era < unregister_era, Error::<T>::NotRegisteredCreator);
+            }
+
+            let current_era = Self::current_era();
+            ensure!(era < current_era, Error::<T>::EraOutOfBounds);
+
+            let staking_info = Self::creator_stake_info(space_id, era).unwrap_or_default();
+            let reward_and_stake =
+                Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
+
+            let (_, stakers_combined_reward_share) =
+                Self::distributed_rewards_between_creator_and_stakers(&staking_info, &reward_and_stake);
+            let staker_reward =
+                Perbill::from_rational(staked, staking_info.total) * stakers_combined_reward_share;
+
+            let mut ledger = Self::ledger(&staker);
+
+            let should_restake_reward = Self::should_restake_reward(
+                restake,
+                creator_info.state,
+                staker_info.latest_staked_value(),
+            );
+
+            if should_restake_reward {
+                staker_info
+                    .stake(current_era, staker_reward)
+                    .map_err(|_| Error::<T>::UnexpectedStakeInfoEra)?;
+
+                // Restaking will, in the worst case, remove one, and add one record,
+                // so it's fine if the vector is full
+                ensure!(
+                    staker_info.len() <= T::MaxEraStakeValues::get(),
+                    Error::<T>::TooManyEraStakeValues
+                );
+            }
+
+            // Withdraw reward funds from rewards holding account
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                staker_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            if should_restake_reward {
```

Comment:
Let's move this line close to mutations:
```suggestion
            let mut ledger = Self::ledger(&staker);
            if should_restake_reward {
```